packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -56,40 +56,56 @@ function getProjectConfiguration(project) {
   return parsedProjectConfiguration;
 }
 
-function _getParsedClassicFilepath(pathSegments, projectConfiguration) {
-  const projectNamePathSeparators = projectConfiguration.pkg.name.split(path.sep);
-
-  pathSegments.splice(
-    0,
-    pathSegments.indexOf(projectNamePathSeparators[0]) + projectNamePathSeparators.length
-  );
-
-  return pathSegments.join(path.sep);
-}
+function _addonFound(filename, projectConfiguration) {
+  const pathSegments = filename.split(path.sep);
+  const addonPaths = projectConfiguration.pkg['ember-addon'].paths;
 
-function _getParsedEmbroiderFilepath(pathSegments) {
-  const RELATIVE_PATH_ROOT = 2;
+  if (!addonPaths) {
+    return;
+  }
 
-  // A typical Embroider pathSegments array would look something like:
-  // ['private', 'var', 'some-hash', 'embroider', 'another-hash', 'tests', 'acceptance', ...]
-  // This strips everything from the first segment up to and including 'another-hash',
-  pathSegments.splice(0, pathSegments.lastIndexOf('embroider') + RELATIVE_PATH_ROOT);
+  const addonNames = addonPaths.map((path) => path.replace('lib/', ''));
 
-  return pathSegments.join(path.sep);
+  return pathSegments.find((addonName) => addonNames.includes(addonName));
 }
 
 /**
  * Get a normalized file path, based on whether the app build is classic or with Embroider
- * @param {object} fileOpts Babel state.file.opts which include root and filename props
+ * @param {object} {root, filename} Babel state.file.opts root & filename
  * @param {object} projectConfiguration Contains project name, ember-addon path info
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
-function getNormalizedFilePath(fileOpts, projectConfiguration) {
-  let { root, filename } = fileOpts;
-  const pathSegments = filename.split(path.sep);
-  const isEmbroider = pathSegments.includes('embroider');
+function getNormalizedFilePath({ root, filename }, projectConfiguration) {
+  const projectPathPrefix = projectConfiguration.pkg.name;
+  const rootDirWithBase = path.join(path.parse(root).dir, path.parse(root).base);
+  const embroiderRegex = new RegExp(`embroider(${path.sep})(.{6})`);
+  const embroiderPrefix = filename.match(embroiderRegex);
+  const addonFound = _addonFound(filename, projectConfiguration);
+
+  if (embroiderPrefix) {
+    const embroiderPrefixIndex = filename.search(embroiderRegex);
+    let embroiderPath = filename.substring(embroiderPrefixIndex);
+
+    if (filename.includes(rootDirWithBase)) {
+      embroiderPath = embroiderPath.replace(rootDirWithBase, '')
+    }
+    embroiderPath = embroiderPath.replace(embroiderPrefix[0] + path.sep, '');
+
+    return embroiderPath;
+  }
+
+  if (addonFound) {
+    const rootDirBaseWithProjectPrefix = path.join(rootDirWithBase, projectPathPrefix);
+    const sourceAddonPath = path.join(rootDirBaseWithProjectPrefix, 'tests', addonFound);
```

Comment:
Trying to understand what we're trying to model here. Could you help me understand what Ember app directory structure we expect this is representing?

Reply:
Here's what they model. I think if I add comments, it will be helpful.

```
    const rootDirBaseWithProjectPrefix = path.join(rootDirWithBase, projectPathPrefix);
    //E.g. /private/var/folders/2j/some-hash/T/tmp-hash/@my-package/app-template

    const sourceAddonPath = path.join(rootDirBaseWithProjectPrefix, 'tests', addonName);
    // E.g. /private/var/folders/2j/some-hash/T/tmp-hash/@my-package/app-template/tests/my-addon

    const targetAddonPath = path.join(rootDirBaseWithProjectPrefix, 'lib', addonName, 'tests');
    // E.g. /private/var/folders/2j/some-hash/T/tmp-hash/@my-package/app-template/lib/my-addon/tests
```

Reply:
Right, but what do they represent in terms of actual, real-world apps. Can you point to any that show this structure?

For instance, it's not clear to me what the first represents. Is it an app with a scope? Why would it have a scope?

For the second, what structure ever has addons nested under a tests directory in the app?

For the third, this represents an approximation of what an app that's using `ember-add-in-repo-tests` would look like, correct?